spec/reek/cli/command/todo_list_command_spec.rb

```diff
@@ -2,67 +2,95 @@
-    let(:configuration) { instance_double 'Reek::Configuration::AppConfiguration' }
-    let(:sources) { [source_file] }
+  let(:smelly_file) do
+    <<-EOS
+      # Smelly class
+      class Smelly
+        # This will reek of UncommunicativeMethodName
+        def x
+          y = 10 # This will reek of UncommunicativeVariableName
+        end
+      end
+    EOS
+  end
 
-    let(:command) do
-      described_class.new(options: options,
-                          sources: sources,
-                          configuration: configuration)
-    end
+  let(:new_configuration_file) do
+    <<-EOS.strip_heredoc
+    # Auto generated by Reeks --todo flag
+    ---
+    detectors:
+      UncommunicativeMethodName:
+        exclude:
+        - Smelly#x
+      UncommunicativeVariableName:
+        exclude:
+        - Smelly#x
+    EOS
+  end
 
-    before do
-      allow(File).to receive(:write).with(described_class::FILE_NAME, String)
+  describe '#execute on smelly source' do
+    around do |example|
+      Dir.mktmpdir('/tmp') do |tmp|
+        Dir.chdir(tmp) do
+          write_smelly_file
+          example.run
+        end
+      end
     end
 
-    context 'when smells are found' do
-      let(:source_file) { SMELLY_FILE }
+    context 'with default configuration file' do
+      let(:default_configuration_file_name) { Reek::Constants::DEFAULT_CONFIGURATION_FILE_NAME }
+      let(:default_configuration) { Reek::Configuration::AppConfiguration.from_path(default_configuration_file_name) }
 
-      it 'shows a proper message' do
-        expected = "\n'.todo.reek' generated! You can now use this as a starting point for your configuration.\n"
-        expect { command.execute }.to output(expected).to_stdout
-      end
+      context 'when does not exist yet' do
+        it 'creates it' do
+          Reek::CLI::Silencer.silently { todo_command.execute }
 
-      it 'returns a success code' do
-        result = Reek::CLI::Silencer.silently { command.execute }
-        expect(result).to eq(Reek::CLI::Status::DEFAULT_SUCCESS_EXIT_CODE)
+          actual_content = File.read(default_configuration_file_name)
+          expect(actual_content).to match(new_configuration_file)
+        end
       end
 
-      it 'writes a todo file with exclusions for each smell' do
-        Reek::CLI::Silencer.silently { command.execute }
-        expected_yaml = {
-          Reek::Configuration::AppConfiguration::DETECTORS_KEY =>
-          { 'UncommunicativeMethodName' => { 'exclude' => ['Smelly#x'] },
-            'UncommunicativeVariableName' => { 'exclude' => ['Smelly#x'] } }
-        }.to_yaml
-        expect(File).to have_received(:write).with(described_class::FILE_NAME, expected_yaml)
-      end
-    end
+      context 'when exists already' do
+        it 'does not update the configuration' do
+          write_existing_configuration default_configuration_file_name
+          command = todo_command configuration: default_configuration
```

Comment:
Does it matter for the test what is passed in `configuration`?

Reply:
Good catch, removed.